pandas/core/reshape/reshape.py

```diff
@@ -1105,6 +1109,179 @@ def get_empty_frame(data) -> DataFrame:
+    ----------
+    data : DataFrame
+        Data which contains dummy-coded variables.
+    sep : str, default None
+        Separator used in the column names of the dummy categories they are
+        character indicating the separation of the categorical names from the prefixes.
+        For example, if your column names are 'prefix_A' and 'prefix_B',
+        you can strip the underscore by specifying sep='_'.
+    base_category : None, Hashable or dict of Hashables, default None
+        The base category is the implied category when a value has non none of the
+        listed categories specified with a one, i.e. if all dummies in a row are
+        zero. Can be a a single value for all variables or a dict directly mapping
+        the base categories to a prefix of a variable.
+
+    Returns
+    -------
+    DataFrame
+        Categorical data decoded from the dummy input-data.
+
+    Raises
+    ------
+    ValueError
+        * When the input `DataFrame` `data` contains NA values.
+        * When the input `DataFrame` `data` contains column names with separators
+          that do not match the separator specified with `sep`.
+        * When a `dict` passed to `base_category` does not include an implied
+          category for each prefix.
+        * When a value in `data` has more than one category assigned to it.
+        * When `base_category=None` and a value in `data` has no category assigned
+          to it.
+    TypeError
+        * When the input `DataFrame` `data` contains non-dummy data.
+        * When the passed `sep` is of a wrong data type.
+        * When the passed `base_category` is of a wrong data type.
+
+    See Also
+    --------
+    :func:`~pandas.get_dummies` : Convert `Series` or `DataFrame` to dummy codes.
+
+    Notes
+    -----
+    The columns of the passed dummy data should only include 1's and 0's,
+    or boolean values.
+
+    Examples
+    --------
+    >>> df = pd.DataFrame({"a": [1, 0, 0, 1], "b": [0, 1, 0, 0],
+    ...                    "c": [0, 0, 1, 0]})
+
+    >>> pd.from_dummies(df)
+    0     a
+    1     b
+    2     c
+    3     a
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 1], "col1_b": [0, 1, 0],
+    ...                    "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                    "col2_c": [0, 0, 1]})
+
+    >>> pd.from_dummies(df, sep="_")
+        col1    col2
+    0    a       b
+    1    b       a
+    2    a       c
+
+    >>> df = pd.DataFrame({"col1_a": [1, 0, 0], "col1_b": [0, 1, 0],
+    ...                    "col2_a": [0, 1, 0], "col2_b": [1, 0, 0],
+    ...                    "col2_c": [0, 0, 0]})
+
+    >>> pd.from_dummies(df, sep="_", base_category={"col1": "d", "col2": "e"})
+        col1    col2
+    0    a       b
+    1    b       a
+    2    d       e
+    """
+    from pandas.core.reshape.concat import concat
+
+    # error: Item "bool" of "Union[Series, bool]" has no attribute "any"
+    if data.isna().any().any():  # type: ignore[union-attr]
```

Comment:
Should you check that data is in fact a DF?  If data is a series then this will produce a cryptic error I think.